File: 1253-sort-the-matrix-diagonally/sort-the-matrix-diagonally.py
class Solution:
    def diagonalSort(self, mat: List[List[int]]) -> List[List[int]]:
        rows = len(mat)
        columns = len(mat[0])
        lookup = {}

        for i in range(rows):
            for j in range(columns):
                cell = mat[i][j]
                diagonal = i - j
                if (diagonal not in lookup):
                    lookup[diagonal] = []
                lookup[diagonal].append(cell)

        for key in lookup:
            lookup[key].sort(reverse=True)

        for i in range(rows):
            for j in range(columns):
                diagonal = (i - j)
                mat[i][j] = lookup[diagonal].pop()

        return mat
                
        
        

File: 1253-sort-the-matrix-diagonally/sort-the-matrix-diagonally.js
var diagonalSort = function(mat) {
    let lookup = {}
    let rows = mat.length
    let columns = mat[0].length

    for (let i = 0; i < rows; i++) {
        for (let j = 0; j < columns; j++) {
            let cell = mat[i][j]
            let diagonal = (i - j)
            if (!(diagonal in lookup)) lookup[diagonal] = []
            lookup[diagonal].push(cell)
        }
    }    

    for (let i = 0; i < rows; i++) {
        for (let j = 0; j < columns; j++) {
            let diagonal = (i - j)
            lookup[diagonal].sort((a, b)=> b - a)
        }
    }

    for (let i = 0; i < rows; i++) {
        for (let j = 0; j < columns; j++) {
            let diagonal = (i - j)
            mat[i][j] = lookup[diagonal].pop()
        }
    }

    return mat
};